bgapi/module.py

```diff
@@ -216,10 +217,18 @@ def procedure_call(self, procedure_type, timeout, throwError=True):
             
             handle = ProcedureCallHandle()
             self._handles[procedure_type] = handle
+
+            # Ensure that the lib does not exceed connection interval.
+            earliest_interval_ts = self.ioTimestamps[procedure_type][0]
+            interval = self.get_procedure_call_interval()
+            while (time.time() - earliest_interval_ts) < interval:
+                time.sleep(max(0.01, time.time() - earliest_interval_ts))
```

Comment:
Yes, you are correct regarding the number of packets being a configurable parameter. Looks like I have been using a rather oversimplified BLE manual when looking into this.

Regarding 'Out of Memory' response.

I am actually using this library to reverse-engineering API for a device (Microbot Push).  Sadly, this implies that I have no access (or insight) into their firmware. Sadly, they drop the connection if central overloads them with orders. And reconnection takes few seconds as to authorize itself with the device by writing couple of values & reading response. By my gut feel, implementing this feature did reduce number of disconnects somewhat.

I have disabled this bit of code in my local checkout and so far I don't see any errors. But it will take me about a week to confirm this as these errors are quite probabilistic.

But from abstract point of view I'd argue that central should abide to the protocol whenever possible. And this includes rate control.